bevy_script_api/src/sub_reflect.rs

```diff
@@ -348,6 +348,13 @@ impl ReflectPath {
             ReflectBase::Component { comp, entity } => {
                 let g = world_ptr.read();
 
+                g.get_entity(*entity).ok_or_else(|| {
+                    ReflectionError::InvalidBaseReference {
+                        base: self.base.to_string(),
+                        reason: "This entity does not exist".to_owned(),
+                    }})?;
+
+
                 let ref_ = self.walk_path(comp.reflect(g.entity(*entity)).ok_or_else(|| {
```

Comment:
Just so we don't look up the entity twice
```suggestion
                let entity_ref = g.get_entity(*entity).ok_or_else(|| {
                    ReflectionError::InvalidBaseReference {
                        base: self.base.to_string(),
                        reason: "This entity does not exist".to_owned(),
                    }})?;
                let ref_ = self.walk_path(comp.reflect(entity_ref).ok_or_else(|| {
```

Reply:
I am assuming you were getting panics in the `g.entity` call right?

Reply:
Right. Good call with not doing the lookup twice, I'll incorporate that